modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3886,7 +3886,7 @@ def n_ary_op(
                 1,
                 new_right_frames,
                 join_type,
-                sort=True,
```

Comment:
Why was `sort=True` here?

Reply:
I think we need to add more tests because you are changing the behavior of a low-level function that is used to implement many high-level functions.

Reply:
My assumption is that someone was thinking this should work for all cases.

Reply:
What tests would you suggest to add?

Reply:
We already have pretty extensive test suite for checking high-level functions.

Reply:
Indeed, handling sort here is a bit tricky. Changed the logic and added a test for some bin op.